lib/app.jsx

```diff
@@ -84,6 +87,28 @@ const isElectron = ( () => {
 	return () => foundElectron;
 } )();
 
+const when = ( condition ) => ( action, fallback = noop ) => ( ... args ) => (
+	condition( ... args ) ? action( ... args ) : fallback( ... args )
+)
+
+const preventDefaultThen = ( fn ) => ( e ) => {
+	e.preventDefault()
+	return fn( e )
+}
+
+const whenCanResize = when( ( { canResize } ) => canResize );
+const whenIsResizing = when( ( { isResizing } ) => isResizing );
+
+const detectCanResize = ( { onResizable, sourceListSelector } ) => ( e ) => {
+	const node = document.querySelector( sourceListSelector )
```

Comment:
@dmsnell do you know of a better way to get a reference to a node in "state-less" way? I'm trying to use a reference to the `.source-list` node to figure out where the right edge of the node is in order to activate resizing.